fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -178,7 +178,58 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
+            if (!scheduleParams.getDisburseDetailMap().isEmpty()) {
+                for (Map.Entry<LocalDate, Money> disburseDetail : scheduleParams.getDisburseDetailMap().entrySet()) {
+                    BigDecimal downPaymentAmount = disburseDetail.getValue().getAmount()
+                            .multiply(loanApplicationTerms.getDisbursedAmountPercentageForDownPayment()).divide(BigDecimal.valueOf(100L));
+                    Money downPayment = Money.of(loanApplicationTerms.getCurrency(), downPaymentAmount);
+                    LoanScheduleModelDownPaymentPeriod installment = LoanScheduleModelDownPaymentPeriod.downPayment(
+                            scheduleParams.getInstalmentNumber(), disburseDetail.getKey(), downPayment,
+                            scheduleParams.getOutstandingBalance());
+
+                    addLoanRepaymentScheduleInstallment(scheduleParams.getInstallments(), installment);
+
+                    scheduleParams.incrementPeriodNumber();
+                    scheduleParams.incrementInstalmentNumber();
+
+                    periods.add(installment);
+
+                    Money newOutstandingBalance = scheduleParams.getOutstandingBalance().minus(downPaymentAmount);
+                    scheduleParams.setOutstandingBalance(newOutstandingBalance);
+
+                    scheduleParams.addTotalCumulativePrincipal(downPayment);
+                    scheduleParams.addTotalRepaymentExpected(downPayment);
+                    totalDownPaymentAmount.add(downPaymentAmount);
+                }
+            } else {
+                BigDecimal downPaymentAmount = loanApplicationTerms.getPrincipal().getAmount()
+                        .multiply(loanApplicationTerms.getDisbursedAmountPercentageForDownPayment()).divide(BigDecimal.valueOf(100L));
+                Money downPayment = Money.of(loanApplicationTerms.getCurrency(), downPaymentAmount);
+                LoanScheduleModelDownPaymentPeriod installment = LoanScheduleModelDownPaymentPeriod.downPayment(
+                        scheduleParams.getInstalmentNumber(), loanApplicationTerms.getExpectedDisbursementDate(), downPayment,
+                        scheduleParams.getOutstandingBalance());
+
+                addLoanRepaymentScheduleInstallment(scheduleParams.getInstallments(), installment);
+
+                scheduleParams.incrementPeriodNumber();
+                scheduleParams.incrementInstalmentNumber();
+
+                periods.add(installment);
+
+                Money newOutstandingBalance = scheduleParams.getOutstandingBalance().minus(downPaymentAmount);
+                scheduleParams.setOutstandingBalance(newOutstandingBalance);
+                scheduleParams.addTotalCumulativePrincipal(downPayment);
+                scheduleParams.addTotalRepaymentExpected(downPayment);
+                totalDownPaymentAmount.add(downPaymentAmount);
+            }
+        }
+
+        while (!scheduleParams.getOutstandingBalance().isEqualTo(totalDownPaymentAmount)
```

Comment:
Was this purposely changed from `!scheduleParams.getOutstandingBalance().isZero()`?

Reply:
fixed